### DISCUSSION TITLE
reading methods that requires contract Type

### DISCUSSION THREAD

**Discussion by nissoh** [wevm/viem#1640] [Question]
hello, i was wondering if this is possible

within a deployed Libary.sol contract i have the following method
````function subscriptionExpiry(IDataStore _dataStore, address _user) public view returns (uint256)````

calling it using readContract
````typescript
readContract({
      abi: abi,
      functionName: 'subscriptionExpiry',
      args: [datastoreAddress, walletAddress]
 })
 ````

i get the following error
`InvalidAbiEncodingType: Type "IDataStore" is not a valid encoding type.
Please provide a valid ABI type. `

**Comment by tmm** [comment]
What does `abi` look like?

**Reply by nissoh** [reply]
hey @tmm the example is simplified, you can check the full abi and code here

the method i have issue with is `function puppetSubscriptionExpiry(IDataStore _dataStore, address _puppet, bytes32 _routeKey) public view returns (uint256)`
https://arbiscan.io/address/0x32b0373B53eC9e16eAFda86b41a6aCf3B2a39f14#code

**Reply by jxom** [reply]
How was the ABI compiled? It looks malformed to me as `IDataStore` is not a valid ABI type.
